File: 이건희/level2/영어 끝말잇기.py
def solution(n, words):
    result = []
    cnt = 0
    word = []

    for i in words:
        cnt += 1
        if len(word) == 0:
            word.append(i)
        elif len(i) == 1:
            break
        else:
            if word[-1][-1] == i[0] and i not in word:
                word.append(i)
            else:
                break

    if len(words) == len(word):
        return [0, 0]
    elif cnt % n == 0:
        return [n, cnt // n]
    else:
        return [cnt % n, cnt // n + 1]

File: 이훈일/Programmers/Level2/영어 끝말잇기.java
import java.util.List;
import java.util.ArrayList;

class Solution {
    public int[] solution(int n, String[] words) {
        List<String> duplication = new ArrayList<String>(){{
            add(words[0]);
        }};
        int[] result = new int[]{0,0};
        char end = words[0].charAt(words[0].length()-1);
        for(int i = 1; i < words.length; i++) {
            if(duplication.contains(words[i])) {
                result[0] = (i+1)%n == 0 ? n : (i+1)%n;
                result[1] = ((i+1)-result[0])/n+1;
                return result;
            }
            if(end != words[i].charAt(0)) {
                result[0] = (i+1)%n == 0 ? n : (i+1)%n;
                result[1] = ((i+1)-result[0])/n+1;
                return result;
            }
            end = words[i].charAt(words[i].length()-1);
            duplication.add(words[i]);
        }
        return result;
    }
}
